apps/arweave/src/ar_mining_worker.erl

```diff
@@ -577,22 +473,47 @@ handle_task({computed_h2, Candidate, _ExtraArgs}, State) ->
 	end,
 	case {PassesDiffChecks, Peer} of
 		{false, _} ->
+			%% h2 does not pass diff checks, do nothing.
 			ok;
-		{_, not_set} ->
+		{Check, not_set} when partial == Check orelse true == Check ->
+			%% This branch only handles the case where we're not part of a coordinated mining set.
+			%% This includes the solo mining setup, and pool mining setup.
+			%% In case of solo mining, the `Check` will alwaysbe `true`.
+			%% In case of pool mining, the `Check` will be `partial` or `true`.
+			%% In either case, we prepare and post the solution.
 			ar_mining_server:prepare_and_post_solution(Candidate);
-		_ ->
+		{Check, _} when partial == Check orelse true == Check ->
+			%% This branch only handles the case where we're part of a coordinated mining set.
+			%% In this case, we prepare the PoA2 and send it to the lead peer.
 			PoA2 = case ar_mining_server:prepare_poa(poa2, Candidate, #poa{}) of
-				{ok, PoA} ->
-					PoA;
+				{ok, PoA} -> PoA;
 				{error, _Error} ->
 					%% Fallback. This will probably fail later, but prepare_poa/3 should
 					%% have already printed several errors so we'll continue just in case.
+					%% df: Is this the right fallback?..
 					#poa{ chunk = Chunk2 }
 			end,
```

Comment:
Perhaps, should move the PoA part into `ar_coordination` here. This is the only place in whole mining worker that touches the PoA logic.

Reply:
agree, I don't like having the poa logic here, makes sense to refactor it into ar_coordination or ar_mining_server

Reply:
Bumping this. I think your point was a good one. But maybe we defer to another PR?